mentat/parsers/block_parser.py

```diff
@@ -0,0 +1,28 @@
+from typing import Any, AsyncGenerator
+
+from typing_extensions import override
+
+from mentat.code_file_manager import CodeFileManager
+from mentat.config_manager import ConfigManager
+from mentat.parsers.file_edit import FileEdit
+from mentat.parsers.original_format.original_format_parsing import (
+    stream_and_parse_llm_response,
+)
+from mentat.parsers.parser import Parser
+from mentat.prompts import block_parser_prompt
+
+
+class BlockParser(Parser):
```

Comment:
It's called block because the changes are given in "blocks" like this?
```
@@start
{
    "file": "core/script.py",
    "action": "insert",
    "insert-after-line": 3,
    "insert-before-line": 4
}
@@code
    if name == "Bob":
        print("Nice to see you again!")
@@end
```

Reply:
Haha that is the reasoning; I don't like the name but I spent too long trying to think of a better name! Definitely let me know if you have any better ideas!

Reply:
The name is fine. I was worried future formats could also be reasonably called "block" but thinking about it it seems like a lot of future formats we will try will be easier to name e.g. GitDiffParser, FullFileParser

Reply:
Hmm, I'm not sure I understand what the File means in this context? Most likely all of our formats will need the model to specify the file.

Reply:
`OriginalFormatParser` would be unambiguous, but `BlockParser` seems fine for now.

The name really just matters if we end up keeping this format around, which we'd only do if it is the best for some models / scenarios. I don't expect that but if it does happen we'll find a name that distinguishes it well from the other formats

Reply:
> Hmm, I'm not sure I understand what the File means in this context?

"File" would mean a literal file on the file system. Technically GitDiffParser would be pulling from a file too :/